### PR TITLE
chore: adaptations for nightly-2024-07-18

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/RingTheory/ClassGroup.lean [hunk 1]
```diff
@@ -384,15 +384,19 @@ of global fields.
...
     rw [Finset.mem_singleton]
     exact ClassGroup.mk_eq_one_iff.mpr (I : FractionalIdeal R⁰ (FractionRing R)).isPrincipal
 
 /-- The class number of a principal ideal domain is `1`. -/
 theorem card_classGroup_eq_one [IsPrincipalIdealRing R] : Fintype.card (ClassGroup R) = 1 := by
   rw [Fintype.card_eq_one_iff]
   use 1
-  refine ClassGroup.induction (R := R) (FractionRing R) (fun I => ?_)
+  #adaptation_note
+  /-- 2024-07-18 need to disable elab_as_elim due to incorrect motive -/
```
Comment (jcommelin) [inline Mathlib/RingTheory/ClassGroup.lean:398]:
```suggestion
  /-- 2024-07-18 need to disable elab_as_elim with `(P := _)` due to type-incorrect motive -/
```

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
bors merge

**Comment by semorrison** [comment]
Ugh, let's just abandoned, it has been superseded by #14942 and `bump/v4.11.0` has already had master merged.

It's not ideal, as we get a less granular history of updates for adaptations, but not terrible.

**Pull request closed without merging**